src/pip/_internal/utils/filesystem.py

```diff
@@ -33,6 +33,8 @@ def check_path_owner(path):
     if sys.platform == "win32" or not hasattr(os, "geteuid"):
         return True
 
+    assert os.path.isabs(path)
```

Comment:
nit: move this to the top of the function, so that it becomes an invariant-on-entry (much more discoverable and easier to understand).